crates/fluvio/src/producer/accumulator.rs

```diff
@@ -66,27 +67,40 @@ impl RecordAccumulator {
         partition_n: PartitionCount,
         compression: Compression,
     ) -> Self {
-        let mut batches = Vec::with_capacity(partition_n as usize);
-        for _ in 0..batches.capacity() {
-            batches.push((BatchEvents::shared(), BatchesDeque::shared()));
+        let mut batches = HashMap::new();
+        for p in 0..partition_n {
+            batches.insert(p, (BatchEvents::shared(), BatchesDeque::shared()));
         }
         Self {
-            batches: Arc::new(batches),
+            batches: Arc::new(RwLock::new(batches)),
             batch_size,
             compression,
             queue_size,
         }
     }
 
+    pub async fn add_partition(
```

Comment:
is this need to be made public?

Reply:
unit test?

Reply:
I think so, but only for crate, I change it.

Reply:
unit test added